### PR TITLE
A minimal `pennylane.about` to generate information about pennylane installations

### PR DESCRIPTION
**Pull request by quantshah**
**Description of the Change:*
This is a simple qutip-style `about` function for `pennylane` which can help a user generate information about the versions of pennylane, scipy, numpy, python installation path etc. 

**Benefits:**
This will make it easier to track the environment a user has while running `pennylane`. I am thinking that in future we could even add something for the hardware backends/plugins available although at the moment I am not sure how to check that. @josh146 any suggestions on what else we could add here.

**Related GitHub Issues:**
#181

### CODE REVIEW

**Comment by quantshah** [comment]
Btw, I am always [blacking the code](https://github.com/ambv/black) so there are smaller changes that happen along with my code. Please ignore them.

File: pennylane/about.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+def about():
+    """
+    Prints the information for pennylane installation.
+    """
+    print("PennyLane is a cross-platform Python library for")
+    print("quantum machine learning, automatic differentiation, and")
+    print("optimization of hybrid quantum-classical computations.")
+    print("")
+    print("Copyright 2018 Xanadu Quantum Technologies Inc.")
```
Comment (josh146) [inline pennylane/about.py]: Just a note to myself that we need to update the date to 2019 everywhere :slightly_smiling_face: 
```suggestion
    print("Copyright 2019 Xanadu Quantum Technologies Inc.")
```

File: pennylane/about.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    print("")
+    print("PennyLane Version:       {}".format(pennylane.__version__))
+    print("Python Version:          {0}.{1}.{2}".format(*sys.version_info[0:3]))
+    print("Platform Info:           {}{}".format(platform.system(), platform.machine()))
+    pennylane_install_path = os.path.dirname(inspect.getsourcefile(pennylane))
+    print("Installation path:       {}".format(pennylane_install_path))
+    print("Numpy Version:           {}".format(numpy.__version__))
+    print("Scipy Version:           {}".format(scipy.__version__))
```
Comment (josh146) [inline pennylane/about.py]: A small comment: we could make use of 
```python
plugin_devices = {entry.name: entry for entry in iter_entry_points("pennylane.plugins")}
```
to also include a list of installed PennyLane devices, as well as plugin version numbers

File: pennylane/about.py (hunk 1 above)
Comment (quantshah) [reply pennylane/about.py]: They should allow `ad infinitum`. I can change all the (c) 2018 to 2019 in one PR.

**Comment by josh146** [comment]
> Btw, I am always [blacking the code](https://github.com/ambv/black) so there are smaller changes that happen along with my code. Please ignore them.

That's fine!

I was curious as to why the coverage had decreased, and then realised this is something that is a bit odd to test. Maybe it's worth adding a small test function to an existing file, just to check that calling `qml.about()` returns a string, and then mark the rest as `# pragma: no cover`

**Comment by quantshah** [comment]
I can write a test using the built-in pytest fixture which checks the stdout and compares it to a string. But to be very strict about it, i.e., to test that all versions for pennylane, scipy etc. are correct, we have to essentially write the same calls that the about function implements in the test and check against the printed version numbers. This seems redundant to me. Nevertheless, I will write a basis str testing test for this.

**Comment by quantshah** [comment]
Phew, this CodeFactor checks are really very minute. Nevertheless, ´pip´can be programmatically called to get details of the package and I used it here. This should be good enough for a start:
´´´
Name: PennyLane
Version: 0.3.1
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: http://xanadu.ai
Author: None
Author-email: None
License: Apache License 2.0
Location: /Users/shahnawaz/Dropbox/dev/pennylane
Requires: numpy, scipy, autograd, toml, appdirs, semantic-version
Required-by: 
Python Version:          3.7.2
Platform Info:           Darwinx86_64
Installed plugins:       ['default.gaussian', 'default.qubit']
Numpy Version:           1.15.0
Scipy Version:           1.2.1
´´´

**Pull request merged**